src/acl.c

```diff
@@ -1909,39 +1903,36 @@ int ACLCheckAllPerm(client *c, int *idxptr) {
     return ACLCheckAllUserCommandPerm(c->user, c->cmd, c->argv, c->argc, idxptr);
 }
 
-/* Check if the user's existing pub/sub clients violate the ACL pub/sub
- * permissions specified via the upcoming argument, and kill them if so. */
-void ACLKillPubsubClientsIfNeeded(user *new, user *original) {
-    /* Do nothing if there are no subscribers. */
-    if (!dictSize(server.pubsub_patterns) &&
-        !dictSize(server.pubsub_channels) &&
-        !dictSize(server.pubsubshard_channels))
-        return;
+int totalSubscriptions(void) {
+    return dictSize(server.pubsub_patterns) +
+           dictSize(server.pubsub_channels) +
+           dictSize(server.pubsubshard_channels);
+}
 
+/* If new can access all channels original return NULL;
+   Otherwise return a list of channels that the new user can access */
+list *getUpcomingChannelList(user* new, user *original) {
     listIter li, lpi;
     listNode *ln, *lpn;
-    robj *o;
-    int kill = 0;
-    
-    /* First optimization is we check if any selector has all channel
-     * permissions. */
+
+    /* Optimization: we check if any selector has all channel permissions. */
     listRewind(new->selectors,&li);
     while((ln = listNext(&li))) {
         aclSelector *s = (aclSelector *) listNodeValue(ln);
-        if (s->flags & SELECTOR_FLAG_ALLCHANNELS) return;
+        if (s->flags & SELECTOR_FLAG_ALLCHANNELS) return NULL;
     }
 
-    /* Second optimization is to check if the new list of channels
+    /* Next, check if the new list of channels
      * is a strict superset of the original. This is done by
      * created an "upcoming" list of all channels that are in
      * the new user and checking each of the existing channels
      * against it.  */
-    list *upcoming = listCreate();
-    listRewind(new->selectors,&li);
-    while((ln = listNext(&li))) {
+    list* upcoming = listCreate();
+    listRewind(new->selectors, &li);
+    while ((ln = listNext(&li))) {
```

Comment:
Unnecessary white space changes
```suggestion
    list *upcoming = listCreate();
    listRewind(new->selectors,&li);
    while((ln = listNext(&li))) {
```